netplan/cli/commands/set.py

```diff
@@ -0,0 +1,119 @@
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+'''netplan set command line'''
+
+import os
+import yaml
+import tempfile
+import shutil
+import re
+
+import netplan.cli.utils as utils
+from netplan.configmanager import ConfigManager
+
+
+class NetplanSet(utils.NetplanCommand):
+
+    def __init__(self):
+        super().__init__(command_id='set',
+                         description='Add new setting by specifying a dotted key=value pair like ethernets.eth0.dhcp4=true',
+                         leaf=True)
+
+    def run(self):
+        self.parser.add_argument('key_value', type=str, help='The nested key=value pair in dotted format')
+        self.parser.add_argument('--origin-hint', type=str, default='90-netplan-set',
+                                 help='Can be used to help choose a name for the overwrite YAML file')
+        self.parser.add_argument('--root-dir', default='/',
+                                 help='Overwrite configuration files in this root directory instead of /')
+
+        self.func = self.command_set
+
+        self.parse_args()
+        self.run_command()
+
+    def command_set(self):
+        if len(self.origin_hint) == 0:
+            raise Exception('Invalid/empty origin-hint')
+        split = self.key_value.split('=', 1)
+        key, value = (split[0], None)
+        if len(split) > 1:
+            value = split[1]
+        if key.startswith('network.'):
+            key = key[8:]
+        self.write_file(key, value, self.origin_hint + '.yaml', self.root_dir)
+
+    def parse_key(self, key, value):
+        # Split at '.' but not at '\.' via negative lookbehind expression
+        split = re.split(r'(?<!\\)\.', 'network.' + key)
+        tree = {}
+        i = 1
+        t = tree
+        for part in split:
+            part = part.replace('\\.', '.')  # Unescape interface-ids, containing dots
+            val = {}
+            if i == len(split):
+                val = value
+            t = t.setdefault(part, val)
+            i += 1
+        return tree
+
+    def merge(self, a, b, path=None):
+        "merges b into a"
+        if path is None:
+            path = []
+        for key in b:
+            if key in a:
+                if isinstance(a[key], dict) and isinstance(b[key], dict):
+                    self.merge(a[key], b[key], path + [str(key)])
+                elif b[key] is None:
+                    del a[key]
+                else:
+                    # Overwrite existing key with new key/value from 'set' command
+                    a[key] = b[key]
+            else:
+                a[key] = b[key]
+        return a
+
+    def write_file(self, key, value, name, rootdir='/', path='etc/netplan/'):
+        tmproot = tempfile.TemporaryDirectory(prefix='netplan-set_')
+        os.makedirs(os.path.join(tmproot.name, 'etc', 'netplan'))
```

Comment:
Ouch.. how did I miss that? Of course we should be using `path` there, instead of defining it manually again.
And yes, it is unused... let me remove that unused argument, but still keep `path` in a variable. This way it won't confuse anybody but can still be changed back quickly, if needed in the future.